Index: src/FixedDataTable.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/FixedDataTable.js	(date 1548896441000)
+++ src/FixedDataTable.js	(revision )
@@ -575,6 +575,7 @@
       this.props.scrollTop !== nextProps.scrollTop ||
       this.props.scrollLeft !== nextProps.scrollLeft) {
       this._didScrollStart();
+      this._didScrollStop();
     }
 
     this._didScrollJump(nextProps);
Index: src/reducers/columnStateHelper.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/reducers/columnStateHelper.js	(date 1548896441000)
+++ src/reducers/columnStateHelper.js	(revision )
@@ -36,10 +36,11 @@
     scrollX = scrollLeft;
   }
 
+  const scrollXAfterScrollLeft = scrollX;
   scrollX = scrollTo(state, props, oldProps.scrollToColumn, scrollX);
 
-  // if scrollX changed due to scrollLeft or scrollToColumn, then set scrollJumpedX
-  const scrollJumpedX = scrollX != state.scrollX;
+  // if scrollX changed due to scrollToColumn, then set scrollJumpedX
+  const scrollJumpedX = scrollX != scrollXAfterScrollLeft;
 
   const { maxScrollX } = columnWidths(state);
   scrollX = Math.min(scrollX, maxScrollX);
