Index: site/webpack-client.config.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- site/webpack-client.config.js	(revision e2b4497a7958f74612e9d1c0da1a77569d62d2a9)
+++ site/webpack-client.config.js	(revision )
@@ -67,7 +67,9 @@
   },
 
   devServer: {
-    host: '0.0.0.0'
+    host: '0.0.0.0',
+    port: '8090',
+    disableHostCheck: true,
   },
 
   plugins: [
